src/consensus/pbft/pbft.h

```diff
@@ -434,6 +434,7 @@ namespace pbft
         {
           gb_info->view_change_list->emplace_back(view, version);
         }
```

Comment:
That's not right, the KV manages the history, including compacts and rollbacks.

Reply:
I think what you are trying to do is ensure that the history is applied when we compact something?
If that is the case then execute pending should be called by compact itself as it would make sense to do that.
Additionally, we do not want to apply everything just the versions that are <= the version you are compacting at.

Reply:
moved to be handled by history